### ISSUE TITLE
Investigate digest comparison logic in DownloadManifest

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#33646]
In the DownloadManifest method, if the manifest is requested via `tag` rather than `digest`, there is no pre-request digest value to use to validate the digest returned from the client.  We instead [compare the digest in the response header to the digest computed by the payload](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs#L502).

We should understand whether these are the intended values to compare in this scenario.

FYI: @pallavit, @lmolkova

**Comment by annelo-msft** [comment]
At a minimum, we could improve this case by comparing to the requested digest in the cases where the `reference` is the digest.

**Comment by annelo-msft** [comment]
@timovv investigated and based on his findings, we decided the following:

>  I did some investigation into how the other clients we know of were handling the digest verification in downloadManifest. Docker CLI appears to be only verifying the digest when an identifier (digest minus the `sha256:`) is provided, and not verifying when it's just a tag, since it can't. ORAS and the Azure CLI don't seem to be doing any validation at all.

We decided we will validate the manifest digest on download similar to what the Docker CLI does.  This is the [recommendation in the REST API spec](https://docs.docker.com/registry/spec/api/#digest-header) and provides a higher level of security for SDK customers.  If the user passes a tag and not a digest (determined by checking if `sha256:` is present), we will still validate the digest from the response header against the content to help detect data corruption.
